e2e/framework/applications.go

```diff
@@ -29,37 +30,50 @@ import (
 // that the application update is available
 func UpdateApplication() {
 	Expect(ConnectToOpsCenter(TestContext.OpsCenterURL, TestContext.ServiceLogin)).To(Succeed())
-	Expect(TestContext.Application.Locator).NotTo(BeNil(), "expected a valid application package")
 
-	nodes := Cluster.Provisioner().NodePool().AllocatedNodes()
-	if len(nodes) == 0 {
-		Failf("expected active nodes in cluster, got none")
-	}
+	if TestContext.Onprem.InstallerURL == "" {
```

Comment:
This if splits the function in two giant unrelated blocks and breaks the reading flow. I'd suggest factoring them out into separate functions w/ descriptive names.